src/kernels/jupyter/jupyterUtils.ts

```diff
@@ -128,8 +130,11 @@ export function createRemoteConnectionInfo(
         dispose: noop,
         rootDirectory: Uri.file(''),
         workingDirectory: serverUri?.workingDirectory,
-        getAuthHeader: serverUri ? () => getJupyterServerUri(uri)?.authorizationHeader : undefined,
-        getWebsocketProtocols: serverUri ? () => getJupyterServerUri(uri)?.webSocketProtocols || [] : () => [],
+        // For remote jupyter servers that are managed by us, we can provide the auth header.
+        // Its crucial this is set to undefined, else password retrieval will not be attempted.
+        getAuthHeader: serverUri && !serverId.startsWith('_builtin') ? () => serverUri?.authorizationHeader : undefined,
```

Comment:
This change is required for things to work with the new kernle picker and remotes